Nodejs/Product/Nodejs/SharedProject/CommonProjectNode.cs

```diff
@@ -431,59 +411,17 @@ private void BoldStartupItem()
             }
         }
 
-        private FileSystemWatcher CreateFileSystemWatcher(string dir)
-        {
-            var watcher = new FileSystemWatcher(dir)
-            {
-                InternalBufferSize = 1024 * 4,  // 4k is minimum buffer size
-                IncludeSubdirectories = true
-                // don't need to set NotifyFilters, 
-                // since the default (LastWrite, FileName, and DirectoryName) works for us
-            };
-
-            // Set Event Handlers
-            watcher.Created += this.FileExistanceChanged;
-            watcher.Deleted += this.FileExistanceChanged;
-            watcher.Renamed += this.FileNameChanged;
-            watcher.Error += this.WatcherError;
-
-            // Delay setting EnableRaisingEvents until everything else is initialized.
-            watcher.EnableRaisingEvents = true;
-
-            return watcher;
-        }
-
-        private FileSystemWatcher CreateAttributesWatcher(string dir)
+        private void StartWatchingFiles()
         {
-            var watcher = new FileSystemWatcher(dir)
-            {
-                IncludeSubdirectories = true,
-                NotifyFilter = NotifyFilters.Attributes,
-                InternalBufferSize = 1024 * 4, // 4k is minimum buffer size
-            };
-
-            // Set Event Handlers
-            watcher.Changed += this.FileAttributesChanged;
-            watcher.Error += this.WatcherError;
+            this.StopWatchingFiles();
```

Comment:
>StopWatchingFiles [](start = 17, length = 17)

Why stop first?  In case more files have come under observation in the meantime?

Reply:
This is called when the project is reloaded for various reasons, and I want to make sure we're not watching any old files.